README.md

```diff
@@ -1 +1,245 @@
+
+## `@interface`: `.h` vs `.m`
+
+You've likely wondered by this point why it is that Objective-C requires *two* files for every class. What's the purpose behind the `.h` and the `.m` files? It sure feels redundant some of the time.
+
+Well, the answer to that is **visibility**. Everything contained in the `.h` header file is what's publicly visible. The `@interface` section is just that: a declaration of the available "interface" for interacting with that class or an instance of it. The interface is, in effect, the "control panel" for that class. It's the set of commands and input/output that's available to the *public*, meaning outside the class's own inner workings. Those inner workings of how a class actually functions is its *implementation* (how it accomplishes what it claims to in its interface) which it should keep *private*. 
+
+Imagine, as an example, an old analog car radio:
+
+![](https://upload.wikimedia.org/wikipedia/commons/thumb/c/c6/Car_Radio_of_Analog_Era.jpg/1280px-Car_Radio_of_Analog_Era.jpg)
+—*Typical car radio of analog era. For Toyota's car, made by Fujitsu Ten.* [Wikipedia Commons](https://commons.wikimedia.org/wiki/File:Car_Radio_of_Analog_Era.jpg)
+
+That front panel with all of the buttons is its *interface*—the public presentation of all the ways that the object is intended to be used. Its *implementation*, however, is kept privately enclosed in a metal box that slides into the dashboard. This is partly for safety (it would be awful to electrocute a passenger!), but mainly for security: the radio manufacturer doesn't want the internal circuitry to get disturbed! That might cause unexpected behavior. 
+
+Providing an interface about public usability via the `.h` header file is how Objective-C classes share information about themselves to other classes. However, *within* that class file's `.m` implementation file, an optional *private interface* can be used for declaring *private properties*. 
+
+The private `@interface` is placed in the `.m` implementation file between the `#import` lines and the `@implementation` section and must be closed with an `@end` keyword before the `@implementation` keyword. Private properties can be added here without making them visible outside of the class. Private *methods*, however, don't need to be declared in the private interface section—the method names in the `@implementation` section make them visible within the current class file.
+
+So, an simple representation of that radio's interface in Objective-C might look something like this:
+
+```objc
+//  FISRadio.h
+
+#import <Foundation/Foundation.h>
+#import <CoreGraphics/CGBase.h>   // imports CGFloat
+
+@interface FISRadio : NSObject
+
+@property (nonatomic) CGFloat amFrequency;
+@property (nonatomic) CGFloat fmFrequency;
+
+@property (nonatomic) CGFloat swBal;
```

Comment:
just trying to represent all the buttons on the pic of the radio. it's totally imperfect since IBActions are methods anyway.